cookbooks/cdo-repository/libraries/git.rb

```diff
@@ -39,7 +39,7 @@ def fetch_updates
         run_opts = run_options(cwd: @new_resource.destination)
         # PATCH: support shallow fetch.
         converge_by("fetch updates for #{@new_resource.remote}") do
-          shell_out!("git fetch #{@new_resource.remote}#{" --depth #{@new_resource.depth}" if @new_resource.depth}", run_opts)
+          shell_out!("git fetch #{@new_resource.remote}", run_opts)
```

Comment:
Note- this change is needed to prevent any existing (non-shallow) deployments from retroactively making their existing repository a shallow-clone (via `git fetch --depth 1`) on their next update.